static/src/javascripts/lib/url.spec.js

```diff
@@ -109,46 +117,49 @@ describe('url', () => {
         window.history.pushState = origPush;
     });
 
-    test('pushQueryString(), replaceQueryString()', () => {
-        ['pushState', 'replaceState'].forEach(method => {
-            const origWindowSearch = window.location.search;
-            const origMethod = window.history[method];
-            const urlMethod = method === 'pushState'
-                ? 'pushQueryString'
-                : 'replaceQueryString';
-            const title = 'new-state-title';
-            const querystring = '/foo/bar';
-            const state = { foo: 'bar' };
-
-            window.history[method] = jest.fn();
-            window.title = title;
-
-            url[urlMethod]({ state, querystring, title });
-            expect(window.history[method]).toHaveBeenCalledWith(
-                state,
-                title,
-                querystring
-            );
-
-            window.history[method].mockClear();
-            url[urlMethod]({ querystring });
-            expect(window.history[method]).toHaveBeenCalledWith(
-                {},
-                title,
-                querystring
-            );
+    const testQueryString = (historyMethod, moduleMethod) => {
+        const origWindowSearch = window.location.search;
+        const origMethod = window.history[historyMethod];
+        const title = 'new-state-title';
+        const querystring = '/foo/bar';
+        const state = { foo: 'bar' };
 
-            Object.defineProperty(window.location, 'search', {
-                writable: true,
-                value: `?${querystring}`,
-            });
+        window.history[historyMethod] = jest.fn();
+        window.title = title;
+
+        moduleMethod({ state, querystring, title });
+        expect(window.history[historyMethod]).toHaveBeenCalledWith(
+            state,
+            title,
+            querystring
+        );
 
-            window.history[method].mockClear();
-            url[urlMethod]({ querystring });
-            expect(window.history[method]).not.toHaveBeenCalled();
+        window.history[historyMethod].mockClear();
+        moduleMethod({ querystring });
+        expect(window.history[historyMethod]).toHaveBeenCalledWith(
+            {},
+            title,
+            querystring
+        );
 
-            window.history[method] = origMethod;
-            window.location.search = origWindowSearch;
+        Object.defineProperty(window.location, 'search', {
+            writable: true,
+            value: `?${querystring}`,
         });
+
+        window.history[historyMethod].mockClear();
+        moduleMethod({ querystring });
+        expect(window.history[historyMethod]).not.toHaveBeenCalled();
+
+        window.history[historyMethod] = origMethod;
+        window.location.search = origWindowSearch;
```

Comment:
```
// TODO: raise another PR to refactor these tests 
```

Tests run in their own sandboxed environment that gets restored at the end of the spec file, so their's no need to restore `window.location.search` or `window.history` in this way